torch/distributed/distributed_c10d.py

```diff
@@ -1295,7 +1303,19 @@ def _new_process_group_helper(
                 pg_options.is_high_priority_stream = False
             pg_options._timeout = timeout
 
-            backend_class = ProcessGroupNCCL(backend_prefix_store, group_rank, group_size, pg_options)
+            # If possible, find a backend to split from by peeling
+            # process group wrappers from the world's default pg.
+            split_from = None
+            if is_initialized():
+                split_from = _world.default_pg._get_backend(_get_pg_default_device())
```

Comment:
Since we are already in the `Backend.NCCL` path, would it be safe to directly use `CUDA` as the device? I could be missing some protection logic in `_get_pg_default_device()` though.